tests/element_handle_test.go

```diff
@@ -1,12 +1,271 @@
 package tests
 
 import (
+	"bytes"
 	_ "embed"
+	"fmt"
+	"image/png"
 	"testing"
 
+	"github.com/dop251/goja"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/xk6-browser/api"
+	"github.com/grafana/xk6-browser/common"
 )
 
+//go:embed static/mouse_helper.js
+var mouseHelperScriptSource string
+
+//nolint:gochecknoglobals
+var htmlInputButton = fmt.Sprintf(`
+<!DOCTYPE html>
+<html>
+  <head>
+	<title>Button test</title>
+  </head>
+  <body>
+	<script>%s</script>
+	<button>Click target</button>
+	<script>
+	  window.result = 'Was not clicked';
+	  window.offsetX = undefined;
+	  window.offsetY = undefined;
+	  window.pageX = undefined;
+	  window.pageY = undefined;
+	  window.shiftKey = undefined;
+	  window.pageX = undefined;
+	  window.pageY = undefined;
+	  window.bubbles = undefined;
+	  document.querySelector('button').addEventListener('click', e => {
+		result = 'Clicked';
+		offsetX = e.offsetX;
+		offsetY = e.offsetY;
+		pageX = e.pageX;
+		pageY = e.pageY;
+		shiftKey = e.shiftKey;
+		bubbles = e.bubbles;
+		cancelable = e.cancelable;
+		composed = e.composed;
+	  }, false);
+	</script>
+  </body>
+</html>
+`, mouseHelperScriptSource)
+
+func TestElementHandleBoundingBoxInvisibleElement(t *testing.T) {
+	p := newTestBrowser(t).NewPage(nil)
+
+	p.SetContent(`<div style="display:none">hello</div>`, nil)
+	element := p.Query("div")
+
+	require.Nil(t, element.BoundingBox())
+}
+
+func TestElementHandleBoundingBoxSVG(t *testing.T) {
+	tb := newTestBrowser(t)
+	p := tb.NewPage(nil)
+
+	p.SetContent(`
+        <svg xmlns="http://www.w3.org/2000/svg" width="500" height="500">
+            <rect id="theRect" x="30" y="50" width="200" height="300"></rect>
+        </svg>`, nil)
```

Comment:
I would do the htmls as following (there are others as well in the PR):

```suggestion
	p.SetContent(`
		<svg xmlns="http://www.w3.org/2000/svg" width="500" height="500">
			<rect id="theRect" x="30" y="50" width="200" height="300"></rect>
		</svg>`,
	nil)
```

I know this PR is not about changing the existing tests. But, while on it, why not?

Reply:
:man_shrugging: Sure, changed in fe52721. I didn't see other instances of this, and I wouldn't want to change this for cases that fit on a single line. Since this used spaces instead of tabs, it makes some sense.

Also, let's not bikeshed about style issues, please.

Reply:
Yeah, it was just a suggestion, not a requirement. Sorry.